src/Polly.Core/Simmy/Behavior/OnBehaviorInjectedArguments.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Polly.Simmy.Behavior;
+
+/// <summary>
+/// Arguments used by the behavior chaos strategy to notify that a custom behavior was injected.
+/// </summary>
+/// <param name="Context">The context associated with the execution of a user-provided callback.</param>
+public readonly record struct OnBehaviorInjectedArguments(ResilienceContext Context);
```

Comment:
We converted all records to regular classes (#1422) to not expose unnecessary APIs. 

(applies to all records in this PR)

Reply:
You can also change this to class (applies basically for all arguments that are used in `ResilienceStrategyTelemetry` since these will boxed anyway)

Reply:
I see that in that PR some were turned into structs and others in classes, so I'm kinda confused 🤔

Reply:
Oh yeah sorry, this space was little turbulent. 

But the final approach is: **Every argument is a struct.**  (avoid using records)

I'll be converting those back to structs :D